mi/match.c

```diff
@@ -178,10 +179,82 @@ mkdtree(Srcloc loc, Node *lbl)
 	t = zalloc(sizeof(Dtree));
 	t->lbl = lbl;
 	t->loc = loc;
-	t->id = ndtree++;
+	t->id = ndtree;
+	lappend(&dtree, &ndtree, t);
 	return t;
 }
 
+static int
+loadeq(Node *a, Node *b)
+{
+	if (a == b)
+		return 1;
+
+	if (exprop(a) != exprop(b))
+		return 0;
+
+	switch (exprop(a)) {
+	case Outag:
+	case Oudata:
+	case Oderef:
+		return loadeq(a->expr.args[0], b->expr.args[0]);
+	case Omemb:
+		return loadeq(a->expr.args[0], b->expr.args[0]) && strcmp(a->expr.args[1]->name.name, b->expr.args[1]->name.name);
+	case Otupget:
+	case Oidx:
+		return loadeq(a->expr.args[0], b->expr.args[0]) && a->expr.args[1]->expr.args[0]->lit.intval == b->expr.args[1]->expr.args[0]->lit.intval;
```

Comment:
I think we have a `liteq()` function you can call from here; take a look at what we do in fold.c

Reply:
Done